src/overlays/actors/ovl_Bg_Spot16_Doughnut/z_bg_spot16_doughnut.c

```diff
@@ -21,15 +30,136 @@ const ActorInit Bg_Spot16_Doughnut_InitVars = {
+        }
+        osSyncPrintf("(ｓｐｏｔ１６ ドーナツ雲)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void BgSpot16Doughnut_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void BgSpot16Doughnut_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+
+    if (!(this->fireFlag & 1)) {
+        this->actor.shape.rot.y -= 0x20;
+        if (this->envColorAlpha < 0xFF) {
+            this->envColorAlpha += 5;
+        } else {
+            this->envColorAlpha = 0xFF;
+        }
+    } else if (globalCtx->csCtx.state != 0 && globalCtx->csCtx.npcActions[2] != NULL &&
+               globalCtx->csCtx.npcActions[2]->action == 2) {
+        if (this->envColorAlpha >= 6) {
+            this->envColorAlpha -= 5;
+        } else {
+            this->envColorAlpha = 0;
+            this->fireFlag &= ~1;
+        }
+    }
+}
+
+// Update function for outwardly expanding and dissipating
+void BgSpot16Doughnut_Update2(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+
+    if (this->envColorAlpha >= 6) {
+        this->envColorAlpha -= 5;
+    } else {
+        Actor_Kill(&this->actor);
+    }
+    this->actor.shape.rot.y -= 0x20;
+    Actor_SetScale(&this->actor, this->actor.scale.x + 0.0019999998f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Doughnut/BgSpot16Doughnut_Update.s")
+void BgSpot16Doughnut_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+    u32 scroll;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[5];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Doughnut/func_808B6680.s")
+    scroll = globalCtx->gameplayFrames & 0xFFFF;
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_spot16_doughnut.c", 210);
+    func_80093D84(globalCtx->state.gfxCtx);
+    if (1) {}
+    gSPMatrix(gfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot16_doughnut.c", 213),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    if (this->fireFlag & 1) {
+        gSPSegment(
+            gfxCtx->polyXlu.p++, 0x08,
+            Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, scroll * (-1), 0, 16, 32, 1, scroll, scroll * (-2), 16, 32));
+        gDPSetEnvColor(gfxCtx->polyXlu.p++, 0xFF, 0x00, 0x00, this->envColorAlpha);
+        gSPDisplayList(gfxCtx->polyXlu.p++, D_06000660);
+    } else {
+        gDPSetEnvColor(gfxCtx->polyXlu.p++, 0xFF, 0xFF, 0xFF, this->envColorAlpha);
+        gDPSetPrimColor(gfxCtx->polyXlu.p++, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF);
+        gSPDisplayList(gfxCtx->polyXlu.p++, D_06000FC0);
+    }
+    Graph_CloseDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_spot16_doughnut.c", 238);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Doughnut/BgSpot16Doughnut_Draw.s")
+// Draw function for outwardly expanding and dissipating
+void BgSpot16Doughnut_Draw2(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
do you think for update2 and draw2 we can give it a name for what it is for? If youre not sure its fine, but based on this comment I think you do know enough